apps/prairielearn/assets/scripts/instructorAssessmentInstancesClient.tsx

```diff
@@ -401,87 +415,229 @@ onDocumentReady(() => {
+                    <option value="add">Add</option>
+                    <option value="subtract">Subtract</option>
+                  </>
+                )
+              ) : null}
+              <option value="set_total">Set total time limit</option>
+              <option value="set_rem">Set remaining time</option>
+              <option value="set_exact">Set exact closing time</option>
+              {row.action === 'set_time_limit_all' ||
+              (row.open && row.time_remaining) !== 'Open (no time limit)' ? (
+                <option value="remove">Remove time limit</option>
+              ) : null}
+              {row.open && row.time_remaining !== 'Expired' ? (
+                <option value="expire">Expire time limit</option>
+              ) : null}
+            </select>
+            <TimeLimitExplanation action={form.action} />
+          </p>
+        ) : null}
+        {showTimeLimitOptions &&
+        form.action !== 'set_exact' &&
+        form.action !== 'remove' &&
+        form.action !== 'expire' ? (
+          <div class="input-group mb-2">
+            <input
+              class="form-control time-limit-field"
+              type="number"
+              name="time_add"
+              aria-label="Time value"
+              value={form.time_add}
+              onChange={(e) => updateFormState('time_add', parseFloat(e.currentTarget.value))}
+            />
+            <span class="input-group-text time-limit-field">minutes</span>
+          </div>
+        ) : null}
+        {showTimeLimitOptions && form.action === 'set_exact' ? (
+          <div class="input-group date-picker mb-2">
+            <input
+              class="form-control date-picker"
+              type="datetime-local"
+              name="date"
+              value={form.date}
+              onChange={(e) => updateFormState('date', e.currentTarget.value)}
+            />
+            <span class="input-group-text date-picker">{timezone}</span>
+          </div>
+        ) : null}
+        {(row.open || !form.reopen_without_limit) &&
+        (form.action === 'set_total' ||
+          form.action === 'set_rem' ||
+          form.action === 'add' ||
+          form.action === 'subtract') ? (
+          <p>Proposed closing time: {proposedClosingTime()}</p>
+        ) : null}
+        <p>
+          {row.has_closed_instance ? (
+            <div class="form-check">
+              <input
+                class="form-check-input"
+                type="checkbox"
+                name="reopen_closed"
+                value="true"
+                checked={form.reopen_closed}
+                id="reopen_closed"
+                onChange={(e) => updateFormState('reopen_closed', e.currentTarget.checked)}
+              />
+              <label class="form-check-label" for="reopen_closed">
+                Also re-open closed instances
+              </label>
+            </div>
+          ) : null}
+        </p>
+        <div class="btn-toolbar">
```

Comment:
Can we right-justify these buttons, in keeping with https://github.com/PrairieLearn/PrairieLearn/pull/10416?